app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php

```diff
@@ -66,6 +66,10 @@ public function execute()
 
                 $orderCommentSender->send($order, $notify, $comment);
 
+                $gridPool = $this->_objectManager
+                    ->create(\Magento\Sales\Model\ResourceModel\GridPool::class);
```

Comment:
Directly using the objectManager, is prohibited. Please read the following [article](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/object-manager.html) for a better understanding. Please use the constructor injection and follow this how to properly add it.
https://github.com/magento/magento2/blob/ed2ffad8f227e253c0aa50393f40478b02310b65/app/code/Magento/Customer/Block/Widget/Dob.php#L82

Reply:
@eduard13 Thanks for the suggestions. I will update the PR.

I have one small doubt here. If you look at tha AddComment.php file, there is one more class created directly using the objectManager which is an existing code. Below is the code snippet.

```
 $orderCommentSender = $this-&gt;_objectManager                    -&gt;create(\Magento\Sales\Model\Order\Email\Sender\OrderCommentSender::class);
                $orderCommentSender-&gt;send($order, $notify, $comment);
```

Do you want me to update this class as well using the constructor injection method?

Reply:
Yes, please.